frontend/src/bundles/common/pages/categories-settings/components/category-item/category-item.tsx

```diff
@@ -0,0 +1,191 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { useCallback, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { categoriesSlice } from '~/bundles/categories/store';
+import { BaseModal, Button } from '~/bundles/common/components/components';
+import {
+    ButtonSize,
+    ButtonType,
+    ButtonVariant,
+    FaIcons,
+} from '~/bundles/common/enums/enums';
+import { findIcon } from '~/bundles/common/helpers/find-icon';
+
+import { Checkbox } from '../checkbox/checkbox';
+import { FormEditCategory } from '../form-create-category/form-edit-category';
+import styles from './styles.module.scss';
+
+type Properties = {
+    id: string;
+    categoryName: string;
+    count?: string | number;
+    type: string;
+    iconKey: string;
+    colorIcon: string;
+};
+
+type RootState = {
+    categories: {
+        checkedCategory: string[];
+    };
+};
+
+const CategoryItem: React.FC<Properties> = ({
+    id,
+    categoryName,
+    count,
+    type,
+    iconKey,
+    colorIcon,
+}) => {
+    const [modalDelete, setModalDelete] = useState(false);
+    const [modalEdit, setModalEdit] = useState(false);
+    // const [editCategory, setEditCategory]= useState({})
+    const dispatch = useDispatch();
+    const isChecked = useSelector<RootState, boolean>((state) =>
```

Comment:
I'm not sure that getting data from redux in this component is a good idea. 
Potentially it may be necessary to use this component on other pages
So, this component should get this information from props.